cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/ZooKeeperRollerTest.java

```diff
@@ -78,6 +80,63 @@ public void testAllPodsAreRolled(VertxTestContext context)  {
                 })));
     }
 
+    @Test
+    public void testNonReadyPodsAreRestartedFirst(VertxTestContext context) {
+
+        Function<Pod, List<String>> shouldRoll = pod -> List.of("Pod was manually annotated to be rolled");
+
+        PodOperator podOperator = mock(PodOperator.class);
+        when(podOperator.isReady(any(), eq("my-cluster-zookeeper-0"))).thenReturn(true);
+        when(podOperator.isReady(any(), eq("my-cluster-zookeeper-1"))).thenReturn(false);
+        when(podOperator.isReady(any(), eq("my-cluster-zookeeper-2"))).thenReturn(true);
+        when(podOperator.listAsync(any(), any(Labels.class))).thenReturn(Future.succeededFuture(PODS));
+
+        ZookeeperLeaderFinder leaderFinder = mock(ZookeeperLeaderFinder.class);
+        when(leaderFinder.findZookeeperLeader(any(), any(), any(), any())).thenReturn(Future.succeededFuture("my-cluster-zookeeper-2"));
+
+        MockZooKeeperRoller roller = new MockZooKeeperRoller(podOperator, leaderFinder, 300_00L);
+
+        Checkpoint async = context.checkpoint();
+        roller.maybeRollingUpdate(Reconciliation.DUMMY_RECONCILIATION, DUMMY_SELECTOR, shouldRoll, new Secret(), new Secret())
+              .onComplete(context.succeeding(v -> context.verify(() -> {
+                  assertThat(roller.podRestarts.size(), is(3));
+                  assertThat(roller.podRestarts.pollFirst(), is("my-cluster-zookeeper-1"));
```

Comment:
Feels like this is quite a subtle to assert the order.

Reply:
Pushed up a refactor to use hamcrest `contains` which gives you a failure like:
```
Expected: iterable containing ["my-cluster-zookeeper-0", "my-cluster-zookeeper-1", "my-cluster-zookeeper-2"]
     but: item 0: was "my-cluster-zookeeper-1"
```
If they're out of order